src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -227,6 +227,11 @@ internal static SyntaxToken GetArgumentModifiers(RefKind refKind)
         {
             var hasBody = !modifiers.IsAbstract && (!modifiers.IsPartial || statements != null);
 
+            if (!hasBody)
+            {
+                modifiers = modifiers - DeclarationModifiers.Async;
+            }
```

Comment:
does this work? (if not, leave as is):
```suggestion
            if (!hasBody)
                modifiers -= DeclarationModifiers.Async;
```

Reply:
works great!